grant/projects/paralink.md

```diff
@@ -0,0 +1,71 @@
+
+3. **Smart XCM Design**
+
+   - **HRMP Channel Smart Detection**
+
+     - Implement **automatic detection of HRMP channels between Polkadot parallel chains** to determine chain pairs eligible for XCM transfers.
+     - Utilize on-chain storage data to retrieve information on all currently active parallel chain channels.
+
+   - **Chain Type and Asset Type Smart Recognition**
+
+     - **Automatically detect whether the target chain is an EVM-compatible chain or a Polkadot chain**, dynamically adjusting the cross-chain transfer logic and process.
+     - **Automatically detect and identify asset types**, determining whether assets are native tokens, synthetic assets, or tokens on EVM chains.
+
+   - **Automated Asset Information Retrieval**
+
+     - **Automatically add supported asset pairs** to ParaLink, supporting a broader range of assets and reducing manual configuration and communication costs with project teams.
+     - Provide querying and filtering functionalities for asset pairs to facilitate users in selecting the assets they wish to transfer.
+     - Reference the [XCM Global Registry](https://github.com/colorfulnotion/xcm-global-registry/tree/action/assets/polkadot) to implement the retrieval and parsing of asset Location information.
+     - Tailor asset information retrieval to accommodate differences in on-chain asset management modules across various chains.
+     - Implement custom handling for **special assets** that cannot be processed automatically, ensuring their support within ParaLink.
+
+   - **Frontend Display Optimization**
+
+     - Handle the display of **Token and Network Logos** on the frontend, prioritizing data acquisition from [Subscan Assets Info](https://github.com/subscan-explorer/assets-info) and using default icons if necessary.
+     - Ensure accurate display of asset and network information on the frontend, providing users with clear information.
+
+   - **Cross-Chain Transfer Logic Enhancement**
+
+     - **Implement cross-chain asset transfer logic between EVM chains and Polkadot chains**, enabling users to transfer assets across different types of chains.
+     - Manage various potential exceptions during the cross-chain process to ensure asset security.
+     - Dynamically adjust the cross-chain transfer logic and process based on automatically detected chain types and asset types.
+
+##### PREPLK Distribution Commitment
+
+To express our gratitude for RingDao's funding support for this project, especially in the early stages, ParaLink has launched an ERC-20 token on the Darwinia mainnet called [PREPLK (Pre ParaLink Token)](https://explorer.darwinia.network/address/0x7ED13f74FD8AE70db03Ae74666d1B443341D8A41), and has committed to distribute 20% of the total supply to the RingDao community. We hope this will help the RingDao community gain more benefits from the project and develop a long-term relationship with ParaLink.
```

Comment:
20% distribution of PREPLK include the support of this grant and other early support previous to this. (Hope to make this point more clear.)

In this way to reflect the invest from RingDAO by holding the PREPLK tokens for IP and governance.

@boundless-forest

Reply:
Done